### PR TITLE
JDK-8324682 Remove redefinition of NULL for XLC compiler

### PR DESCRIPTION
**Pull request by suchismith1993**
Remove redefinition of NULL for XLC compiler
In globalDefinitions_xlc.hpp there is a redefinition of NULL to work around an issue with one of the AIX headers (<dirent.h>). 
Once all uses of NULL in HotSpot have been replaced with nullptr, there is no longer any need for this, and it can be removed.

JBS ISSUE : [JDK-8324682](https://bugs.openjdk.org/browse/JDK-8324682)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8324682](https://bugs.openjdk.org/browse/JDK-8324682): Remove redefinition of NULL for XLC compiler (**Enhancement** - P4)

### Reviewers
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)
 * [Matthias Baesken](https://openjdk.org/census#mbaesken) (@MBaesken - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18064/head:pull/18064` \
`$ git checkout pull/18064`

Update a local copy of the PR: \
`$ git checkout pull/18064` \
`$ git pull https://git.openjdk.org/jdk.git pull/18064/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18064`

View PR using the GUI difftool: \
`$ git pr show -t 18064`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18064.diff">https://git.openjdk.org/jdk/pull/18064.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18064#issuecomment-1971226556)

### CODE REVIEW

File: src/hotspot/share/utilities/globalDefinitions_xlc.hpp [hunk 1]
```diff
@@ -94,21 +94,7 @@
 // NULL vs NULL_WORD:
 // Some platform/tool-chain combinations can't assign NULL to an integer
 // type so we define NULL_WORD to use in those contexts. For xlc they are the same.
```
Comment (kimbarrett) [inline src/hotspot/share/utilities/globalDefinitions_xlc.hpp:96]: The last sentence of the comment is no longer true.

**Review by kimbarrett — CHANGES_REQUESTED** [review]
Remember to update copyrights in the changed files.

File: src/hotspot/share/utilities/globalDefinitions_xlc.hpp [hunk 2]
```diff
@@ -1,6 +1,7 @@
 /*
  * Copyright (c) 1998, 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (RealCLanger) [inline src/hotspot/share/utilities/globalDefinitions_xlc.hpp:2]: You need to update the year for the Oracle copyright. Adding additional copyright lines, such as IBM, is usually only done when you substantially change a file, i.e. add larger non-trivial portions of code.

**Review by RealCLanger — CHANGES_REQUESTED** [review]

File: src/hotspot/share/utilities/globalDefinitions_xlc.hpp (hunk 2 above)
Comment (suchismith1993) [reply src/hotspot/share/utilities/globalDefinitions_xlc.hpp]: Ok. Thanks will keep that in mind. Updated the copyrights year.

**Review by MBaesken — APPROVED** [review]

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Comment by kimbarrett** [comment]
> /integrate

For HotSpot changes, at least 2 reviewers (at least one a Reviewer) are required.  The Skara tooling doesn't know about
that policy for HotSpot (and some other areas that have similar policies), so will mark a PR ready even though that
requirement hasn't been satisfied.  Sometimes someone will use the /reviewers command to enforce the policy, but
since it's a "well known" policy that often isn't bothered with.

That said, since I've now also approved it, I will also sponsor.

**Pull request closed without merging**